cpp/src/backends/torch_scripted/handler/base_handler.cc

```diff
@@ -37,11 +37,26 @@ void BaseHandler::Handle(
   std::pair<std::string&, std::map<uint8_t, std::string>&> idx_to_req_id(
       req_ids, map_idx_to_req_id);
   try {
+    auto start_time = std::chrono::high_resolution_clock::now();
     auto inputs =
         Preprocess(device, idx_to_req_id, request_batch, response_batch);
     auto outputs =
         Inference(model, inputs, device, idx_to_req_id, response_batch);
     Postprocess(outputs, idx_to_req_id, response_batch);
+    auto stop_time = std::chrono::high_resolution_clock::now();
+    std::chrono::duration<double, std::milli> duration = stop_time - start_time;
+    try {
+      auto& handler_time_metric =
+          torchserve::MetricsRegistry::GetMetricsCacheInstance()->GetMetric(
+              torchserve::MetricType::GAUGE, "HandlerTime");
+      handler_time_metric.AddOrUpdate(
+          std::vector<std::string>{manifest_->GetModel().model_name, "Model"},
+          idx_to_req_id.first, duration.count());
+    } catch (std::runtime_error& e) {
+      TS_LOG(ERROR, e.what());
+    } catch (std::invalid_argument& e) {
+      TS_LOGF(ERROR, "Failed to record HandlerTime metric. {}", e.what());
+    }
```

Comment:
I checked benchmark code which needs predictiontime to generate model latency.

In fact, predictiontime is as same as handlertime. To make it simple, please move this to https://github.com/pytorch/serve/blob/cpp_backend/cpp/src/backends/torch_scripted/torch_scripted_backend.cc#L77 and emit 2 metric at there.

Reply:
Updated implementation and opened PR: https://github.com/pytorch/serve/pull/2012